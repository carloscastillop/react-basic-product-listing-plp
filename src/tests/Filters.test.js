import React from 'react';
import ReactDOM from 'react-dom';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

import Filter from '../components/Filters';

afterEach(cleanup);
/**
 * Basic render
 */
it('Filter renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Filter/>, div);
});

/**
 * Render with colour filters example (A)
 */
it('Filter renders filter correctly', () => {
    const colours = ['Red', 'Green', 'Blue'];
    const {getByTestId} = render(<Filter colours={colours}/>)
    expect(getByTestId('colourFilterSelect')).toHaveTextContent('Filter by colour', 'Red', 'Green', 'Blue')
});

/**
 * Render with no colour filters (B)
 */
it('Filter renders filter with empty colours', () => {
    const colours = [];
    const {getByTestId} = render(<Filter colours={colours}/>)
    expect(getByTestId('colourFilterSelectEmpty')).toHaveTextContent('No colour filters available')
});

/**
 * Snapshot (A)
 */
it('Filter matches snapshot A', () => {
    const colours = ['Red', 'Green', 'Blue'];
    const tree = renderer.create(<Filter colours={colours}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

/**
 * Snapshot (B)
 */
it('Filter matches snapshot empty filter B', () => {
    const colours = [];
    const tree = renderer.create(<Filter colours={colours}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
