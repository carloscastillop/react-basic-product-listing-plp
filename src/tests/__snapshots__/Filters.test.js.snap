// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter matches snapshot A 1`] = `
<form
  className="my-4"
>
  <div
    className="container"
  >
    <div
      className="form-group"
    >
      <label
        className="sr-only"
        htmlFor="colourFilterSelect"
      >
        Colour filter
      </label>
      <select
        className="form-control"
        data-testid="colourFilterSelect"
        id="colourFilterSelect"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Filter by colour
        </option>
        <option
          value="Red"
        >
          Red
        </option>
        <option
          value="Green"
        >
          Green
        </option>
        <option
          value="Blue"
        >
          Blue
        </option>
      </select>
    </div>
  </div>
</form>
`;

exports[`Filter matches snapshot empty filter B 1`] = `
<form
  className="my-4"
>
  <div
    className="container"
  >
    <div
      className="form-group"
    >
      <label
        className="col-form-label sr-only"
        data-testid="colourFilterSelectEmpty"
      >
        No colour filters available
      </label>
    </div>
  </div>
</form>
`;
